firebase-installations/src/main/java/com/google/firebase/installations/FirebaseInstallations.java

```diff
@@ -451,4 +477,35 @@ private Void deleteFirebaseInstallationId() throws FirebaseInstallationsExceptio
     persistedInstallation.insertOrUpdatePersistedInstallationEntry(entry.withNoGeneratedFid());
     return null;
   }
+
+  private static boolean isIIDVersionCompatible() {
+    try {
+      final Class<FirebaseInstanceId> firebaseInstanceIdClass = FirebaseInstanceId.class;
```

Comment:
Proguard may end up removing this line due to the `code/removal/variable` optimization, I would suggest you test with this optimization as well as with `code/removal/advanced`, `code/removal/simple` optimizations.

Reply:
+1

Reply:
Done. PTAL.

Reply:
Could you explain how it was done? Can we have an integ test for this?